source/logrepl/combined_iterator.go

```diff
@@ -0,0 +1,59 @@
+package logrepl
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/jackc/pgx/v4"
+)
+
+const (
+	TypeSnapshot int = iota
+	TypeCDC
+)
+
+// CombinedIterator binds a CDCIterator and a SnapshotIterator together for
+// coordinating the hand-off from snapshot to CDC operation modes.
+type CombinedIterator struct {
```

Comment:
Might even be called just `Iterator`, it's the only iterator that will be used by the source so `logrepl.Iterator` looks nice IMO.